### PR TITLE
Fix format problem in composite of unmapped

### LINKED ISSUE #50600: composite and terms aggregations behave differently on `ip` fields
TL;DR `composite` aggs behave differently than `terms` agg when there is an index in the pattern that matches the prefix of the other indices that does NOT have a mapped field matching the aggregated field. They behave the same if the "prefix index" has the field, but it is indexed as the wrong type.

Assume we have the following indices
```
PUT test_index-0001
{
  "mappings": {
    "properties": {
      "ip": {
        "type": "ip"
      }
    }
  }
}

PUT test_index-0002
{
  "mappings": {
    "properties": {
      "ip": {
        "type": "ip"
      }
    }
  }
}

PUT test_index-0001/_doc/1
{
  "ip": "10.2.1.1"
}
PUT test_index-0002/_doc/1
{
  "ip": "10.1.1.2"
}
```

Lets add a new index that does not have an `ip` field
```
PUT test_index-0003
{
  "mappings": {
    "properties": {
      "somefield": {
        "type": "ip"
      }
    }
  }
}
```

Composite aggs behaves as follows (as of 7.5).
```
GET test_index-0001,test_index-0002,test_index*/_search?size=0
{
  "aggs": {
    "composite_ips": {
      "composite": {
        "sources": [
          {
            "ip": {
              "terms": {
                "field": "ip"
              }
            }
          }
        ]
      }
    }
  }
}
>
"aggregations" : {
    "composite_ips" : {
      "after_key" : {
        "ip" : "10.2.1.1"
      },
      "buckets" : [
        {
          "key" : {
            "ip" : "10.1.1.2"
          },
          "doc_count" : 1
        },
        {
          "key" : {
            "ip" : "10.2.1.1"
          },
          "doc_count" : 1
        }
      ]
    }
  }
```

But if the index that is added is instead:
```
PUT test_index
{
  "mappings": {
    "properties": {
      "somefield": {
        "type": "ip"
      }
    }
  }
}
```
The following is returned from the same composite agg (image for readability)
image

Note the difference with a plain `terms` agg
```
GET test_index-0001,test_index-0002,test_index*/_search?size=0
{
  "aggs": {
    "term_ips": {
      "terms": {
        "field": "ip"
      }
    }
  }
}
>
"aggregations" : {
    "term_ips" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [
        {
          "key" : "10.1.1.2",
          "doc_count" : 1
        },
        {
          "key" : "10.2.1.1",
          "doc_count" : 1
        }
      ]
    }
  }
```

NOTE: if the `test_index` has the following mapping, composite aggs and terms behave the ***SAME*** (showing raw bytes instead of human readable strings)
```
DELETE test_index
PUT test_index
{
  "mappings": {
    "properties": {
      "ip": {
        "type": "ip"
      }
    }
  }
}

GET test_index-0001,test_index-0002,test_index*/_search?size=0
{
  "aggs": {
    "term_ips": {
      "terms": {
        "field": "ip"
      }
    }
  }
}
```
(image for readability)
image

Related issues/PRs:
*  https://github.com/elastic/elasticsearch/issues/17705
* https://github.com/elastic/elasticsearch/pull/18408
* https://github.com/elastic/elasticsearch/pull/18003

### PR DESCRIPTION
**Pull request by nik9000**
When a composite aggregation is reduced using the results from an
index that has one of the fields unmapped we were throwing away the
formatter. This is mildly annoying, except in the case of IP addresses
which were coming out as non-utf-8-characters. And tripping assertions.

This carefully preserves the formatter from the working bucket.

Closes #50600

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 1]
```diff
@@ -335,6 +340,11 @@ public String format(BytesRef value) {
         public BytesRef parseBytesRef(String value) {
             return new BytesRef(InetAddressPoint.encode(InetAddresses.forString(value)));
         }
+
+        @Override
+        public String toString() {
+            return "ip";
+        }
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java]: I added these to help with debugging and figure they are probably fine to keep.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/InternalCompositeTests.java [hunk 2]
```diff
@@ -249,6 +251,30 @@ public void testReduceSame() throws IOException {
         }
     }
 
+    /**
+     * Check that reducing with an unmapped index produces useful formats.
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/composite/InternalCompositeTests.java]: Thank you for including a note of what we're testing, helps so much when debugging test failures!

**Review by not-napoleon — APPROVED** [review]
Nice clean fix, LGTM.

**Review by jimczi — APPROVED** [review]
LGTM too, thanks @nik9000 !

**Comment by nik9000** [comment]
Thanks @not-napoleon and @jimczi !

**Pull request merged**